src/autowiring/AnySharedPointer.h

```diff
@@ -68,6 +68,8 @@ struct AnySharedPointer {
     return obj && try_assign(obj);
   }
 
+  const std::shared_ptr<void>& as_void(void) const { return m_ptr; }
+
```

Comment:
Quite correct.  Actually, the main motivation to using `AnySharedPointer` is not necessarily that the code becomes any simpler, but that I can avoid using templated functions altogether and move to runtime versions of the same.  These runtime versions are compiled once and don't increase build times downstream.

This particular routine is mainly intended to make unit tests a little easier to write in that sometimes I don't actually care about the underlying type, I just want whatever the pointed-to object is.  There are implicit dangers of doing this--in particular:

```C++
std::shared_ptr<Derived> p;
AnySharedPointer x = std::static_pointer_cast<Base>(p);
ASSERT_EQ(p.get(), x.as_void());  // Might fail!
```